### ISSUE TITLE
how do you configure JPAStreamer in gradle if your JPA model is written in Kotlin?

### ISSUE THREAD

**Issue by SchlauFuchs** [speedment/jpa-streamer#213]
Hi!

It seems like if the JPA classes are written in Kotlin, adding the plugin to gradle does not result in generated sources. what do I need to do here?

**Comment by minborg** [comment]
The generation of the support classes is done using an annotation processor. 

Perhaps this can provide some guidelines : https://kotlinlang.org/docs/ksp-overview.html

**Comment by SchlauFuchs** [comment]
Yes, I understood that with the annotation processor, and I added the
processor dependency to the gradle build script according to the
instructions. Still, I do not see anything trying to create generated code
during the build - do you know if the JpaStreamer annotation processor
interacts with the KSP?

**Comment by minborg** [comment]
The annotation processor is only tested using Java. I am not a Kotlin expert but maybe the stage whereby Java code is inspected in the annotation processor does not exist during a Kotlin build? In that case, we should have to add some explicit support for KSP and Kotlin. Would it be possible for you to write one of your `@Entity` classes in Java and see if that works? I understand that this is not an ideal solution but this would perhaps allow a work-around?

**Comment by SchlauFuchs** [comment]
I cannot turn them to Java as it is company property and policy is to write
Kotlin code... but I might have some time experimenting about the
annotation processing in our IP sprint which comes in a few weeks.

**Comment by minborg** [comment]
Understood. Let's check back in a few weeks and see.

**Comment by Leokao4Autopass** [comment]
I have a same problem, I want use jpa-streamer with kotlin project. 
any progress on this discussion? 

update:
I successful use it with kapt in quarkus

build.gradle.kts
```
plugins {
    kotlin("jvm") version "1.9.21"
    kotlin("plugin.allopen") version "1.9.21"
    kotlin("plugin.jpa") version "1.9.21"
    id("io.quarkus")
    kotlin("kapt") version "1.9.21"
}

dependencies {
    kapt("com.speedment.jpastreamer:fieldgenerator-standard:3.0.4")
}
```
